src/NServiceBus.Core/Licensing/ConsoleLicensePrompt.cs

```diff
@@ -0,0 +1,146 @@
+                var input = Console.ReadKey();
+                Console.WriteLine();
+
+                if (int.TryParse(input.KeyChar.ToString(), out var optionIndex))
+                {
+                    if (optionIndex > 0 && optionIndex <= options.Count)
+                    {
+                        var (result, license) = options[optionIndex - 1].optionAction();
+
+                        if (!string.IsNullOrWhiteSpace(result))
+                        {
+                            Console.WriteLine(result);
+                            Task.Delay(TimeSpan.FromSeconds(2)).GetAwaiter().GetResult();
+                        }
+
+                        if (license != null)
+                        {
+                            return license;
+                        }
+                    }
+                }
+            }
+        }
+
+        static void ListOptions(License trialLicense, List<(string optionText, Func<(string result, License license)> optionAction)> options)
+        {
+            Console.Clear();
+
+            Console.ForegroundColor = ConsoleColor.Red;
+            Console.WriteLine(trialLicense.IsExtendedTrial ? "Your extended trial license has expired!" : "Your trial license has expired!");
+            Console.ResetColor();
+
+            Console.WriteLine("Press:");
+
+            for (var i = 0; i < options.Count; i++)
+            {
+                Console.WriteLine($"{i + 1}: {options[i].optionText}");
+            }
+        }
+
+        static (string result, License license) ImportLicense()
+        {
+            Console.Clear();
+
+            Console.WriteLine("Specify the path to your license file and press [Enter]:");
+            var input = Console.ReadLine();
+
+            try
+            {
+                if (!Path.IsPathRooted(input))
+                {
+                    input = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, input);
+                }
+
+                if (!File.Exists(input))
+                {
+                    return ($"No file found at '{input}'", null);
+                }
+
+                Console.WriteLine("Validating license file...");
+                var licenseText = File.ReadAllText(input);
+
+                if (!LicenseVerifier.TryVerify(licenseText, out var licenseVerifactionException))
+                {
+                    return ("Specified file does not contain a valid license: " + licenseVerifactionException.Message, null);
+                }
+
+                Console.WriteLine("Importing license...");
+                var providedLicense = LicenseDeserializer.Deserialize(licenseText);
+
+                if (LicenseExpirationChecker.HasLicenseExpired(providedLicense))
+                {
+                    return ("Imported license has expired. Please provide a valid license.", null);
+                }
+
+                // Store licenses locally in the License subfolder which will be probed as a license source
```

Comment:
Putting this here doesn't really make sense, since this will only happen during development, which means it will get wiped if the bin folder is deleted. To make this really work, we'd need to add this as Content to the project, which we wouldn't want to try to do.

I think this demonstrates that we need to define a global file system license source that we can use instead.